mlos_bench/mlos_bench/storage/base_storage.py

```diff
@@ -209,6 +230,9 @@ def _setup(self) -> None:
 
             This method is called by `Storage.Experiment.__enter__()`.
             """
+            self._status = Status.RUNNING
+            self._driver_name = platform.node()
+            self._driver_pid = os.getpid()
```

Comment:
These seem ok to initialize the values from `None`, but might be problematic for resuming an Experiment.

There are a few cases I can think of:
1. An individual `mlos_bench` driver process fails and needs to be restarted.
2. The `mlos_benchd` service dies, but the `mlos_bench` process is still running.
3. The whole driver host fails and either needs to be restarted on that same backend or else picked up by a new one (for simplicity, let's assume we only support the former for now.  We can add a heartbeat mechanism later to support the latter).